keras/backend/tensorflow_backend.py

```diff
@@ -859,7 +859,8 @@ def set_value(x, value):
     '''Sets the value of a tensor variable,
     from a Numpy array.
     '''
-    tf_dtype = _convert_string_dtype(dtype(value))
+    value = np.asarray(value)
+    tf_dtype = _convert_string_dtype((x*1.).dtype)
```

Comment:
Well, actually x.dtype returns a special tensorflow "ref" type that is not compatible with Keras and creates a bug. It is a similar problem as in the dropout function (in the same file), the *1. dummy multiplication is a simple workaround (but quite strange I agree)...